lib/node_modules/@stdlib/string/base/replace-before/test/test.js

```diff
@@ -0,0 +1,202 @@
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			replaceBefore( 'beep boop', value, 'foo' );
+		};
+	}
+});
+
+tape( 'the function throws an error if not provided a string primitive as its third argument', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		5,
+		NaN,
+		null,
+		void 0,
+		true,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			replaceBefore( 'beep boop', 'beep', value );
+		};
+	}
+});
+
+tape( 'the function replaces the substring before the first occurrence of a specified search string', function test( t ) {
+	var expected;
+	var actual;
+	var str;
+    var replacement;
+
+	str = 'beep boop';
+    replacement = 'foo';
+	actual = replaceBefore( str, ' ' , replacement );
+	expected = 'foo boop';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	str = 'beep boop';
+    replacement = 'foo';
+	actual = replaceBefore( str, 'p', replacement );
+	expected = 'foop boop';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	str = 'Hello, World!';
+    replacement = 'foo';
+	actual = replaceBefore( str, 'o', replacement );
+	expected = 'fooo, World!';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns the substring before a provided search string (Unicode characters)', function test( t ) {
+	var expected;
+	var actual;
+	var str;
+    var replacement;
+
+	str = 'beep 😀 boop 😀 baz';
+    replacement = 'foo';
+	actual = replaceBefore( str, '😀', replacement );
+	expected = 'foo😀 boop 😀 baz';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	str = '🤖 Robot army 🤖!';
+    replacement = 'foo';
+	actual = replaceBefore( str, '🤖', replacement );
+	expected = 'foo🤖 Robot army 🤖!';
+
+	str = '🐺 Wolf brothers 🐺';
+    replacement = 'foo';
+	actual = replaceBefore( str, 'o', replacement );
+	expected = 'fooolf brothers 🐺';
+
+	t.end();
+});
+
+tape( 'the function returns the entire string if the search string is not found', function test( t ) {
+	var expected;
+	var actual;
+	var str;
+    var replacement;
+
+	str = 'beep boop';
+    replacement = 'foo';
+	actual = replaceBefore( str, 'z', replacement );
+	expected = 'foobeep boop';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	str = 'beep boop';
+    replacement = 'foo';
+	actual = replaceBefore( str, 'baz', replacement );
+	expected = 'foobeep boop';
+	t.strictEqual( actual, expected, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns the replacement string if the search string is the empty string', function test( t ) {
+	var expected;
+	var actual;
+	var str;
+    var replacement;
```

Comment:
Variable order. Make sure you have run `make init` before committing. Doing so will ensure that linters are run, which should have caught this lint error.